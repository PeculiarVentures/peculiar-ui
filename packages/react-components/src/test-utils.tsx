import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
