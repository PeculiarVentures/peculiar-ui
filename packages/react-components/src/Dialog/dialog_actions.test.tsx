import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { DialogActions } from './index';

describe('<DialogActions />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(
      <DialogActions>
        Buttons
      </DialogActions>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(
      <DialogActions className="my-class-name">
        Buttons
      </DialogActions>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
