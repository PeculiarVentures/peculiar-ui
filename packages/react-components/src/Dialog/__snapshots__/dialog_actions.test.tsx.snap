// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogActions /> should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: var(--pv-size-base-3) var(--pv-size-base-4) var(--pv-size-base-4);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  gap: var(--pv-size-base-2);
}

<footer
    class="emotion-0 my-class-name"
    data-key="dialog.actions"
  >
    Buttons
  </footer>
</DocumentFragment>
`;

exports[`<DialogActions /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: var(--pv-size-base-3) var(--pv-size-base-4) var(--pv-size-base-4);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  gap: var(--pv-size-base-2);
}

<footer
    class="emotion-0"
    data-key="dialog.actions"
  >
    Buttons
  </footer>
</DocumentFragment>
`;
